linkerd/core/src/main/scala/io/buoyant/linkerd/AnnouncerInitializer.scala

```diff
@@ -0,0 +1,12 @@
+package io.buoyant.linkerd
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.twitter.finagle.Announcer
+import io.buoyant.config.ConfigInitializer
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "kind")
+trait AnnouncerConfig {
+  def mk(): Announcer
+}
+
+trait AnnouncerInitializer extends ConfigInitializer
```

Comment:
In your example are you trying to announce the same server to two different names, foo and bar?

If registering multiple servers I'd expect the config to be
```
announcers:
- kind: io.l5d.whatever
  path: /role/env
servers:
- label: foo
  ...
- label: bar
  ...
```